### PR TITLE
Stats: Add tool tip to date range picker

### PR DESCRIPTION
**Pull request by annacmc**
Related to https://github.com/Automattic/red-team/issues/216

## Proposed Changes

* Add a tooltip to the daterangepicker trigger button

## Why are these changes being made?

* As part of the header refresh in the Date Filtering for the Traffic Page project

## Testing Instructions

* Open Calypso Live Branch
* Apply feature flag `stats/new-date-filtering`
* Confirm that the daterange picker button now has a tooltip, which looks like the screenshot

image

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/components/stats-date-control/index.tsx [hunk 1]
```diff
@@ -166,17 +167,19 @@ const StatsDateControl = ( {
...
-								const event_from = isOdysseyStats ? 'jetpack_odyssey' : 'calypso';
-								recordTracksEvent( eventNames[ event_from ][ 'trigger_button' ] );
-								onTriggerClick();
-							} }
-							ref={ buttonRef }
-						>
-							{ getButtonLabel() }
-							<Icon className="gridicon" icon={ calendar } />
-						</Button>
+						<Tooltip text={ translate( 'Filter all data by date' ) }>
```
Comment (dognose24) [inline client/components/stats-date-control/index.tsx]:
```suggestion
						<Tooltip text={ translate( 'Filter all data by date' ) } placement="bottom-end">
```
The tooltip position on the design seems to be the `bottom-end`.

|Before|After|
|-|-|
|截圖 2024-10-28 下午1 17 10|截圖 2024-10-28 下午1 16 02|

**Review by dognose24 — COMMENTED** [review]
The applied tooltip always shows, regardless of the feature flag, which might not fit the current behavior. We must also consider not directly using the default text since the `StatsDateControl` is a shared component.

**Comment by annacmc** [comment]
> The applied tooltip shows, regardless of the feature flag, which might not fit the current behavior.

Hmm interesting point. I had chosen not to put this behind a feature flag, because it feels useful and separate - but the more I think about it, the more it makes sense in both cases. It is part of the header refresh task (of which the rest is feature flagged) so I think you're right and it makes sense to have it the same. Thanks 🙏

**Comment by dognose24** [comment]
> because it feels useful and separate

I think so. So maybe adjusting the default text is enough rather than gate the tooltip with a feature flag. 🤔

**Review by dognose24 — APPROVED** [review]
Works as described! 👍🏼 I imagined passing a prop `tooltipText` defaulted to empty, which could also receive the string determined by the feature flag outside. However, we could iterate the component to be versatile when refactoring after flipping the flag.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16950409

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @annacmc for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16950409).

**Pull request merged**